milla/src/lib.rs

```diff
@@ -0,0 +1,2231 @@
+                "###", //
+            ],
+            |c: char| match c {
+                'Y' => Air(Box::new([
+                    (ATMOS_OXYGEN, 8.0),    //
+                    (ATMOS_NITROGEN, 14.0), //
+                    // This value is arbitrary, it's just what we get right now.
+                    // Update it if the calculations changed intentionally.
+                    (ATMOS_THERMAL_ENERGY, 8_800.0),
+                ])),
+                '#' => Wall,
+                _ => Space,
+            },
+            test_z,
+        );
+    }
+
+    // Air should exchange with lavaland atmosphere.
+    #[test]
+    fn tick_affected_by_lavaland_atmosphere() {
+        let test_z = 11;
+
+        set_pattern(
+            &[
+                "###", //
+                "#X#", //
+                "###", //
+            ],
+            |c: char| match c {
+                'X' => Air(Box::new([
+                    (ATMOS_OXYGEN, 0.0),                      //
+                    (ATMOS_NITROGEN, 0.0),                    //
+                    (ATMOS_MODE, ATMOS_MODE_LAVALAND as f32), //
+                ])),
+                '#' => Wall,
+                _ => Space,
+            },
+            test_z,
+        );
+
+        tick();
+
+        expect_pattern(
+            &[
+                "###", //
+                "#Y#", //
+                "###", //
+            ],
+            |c: char| match c {
+                'Y' => Air(Box::new([
+                    // Gain 50% of Lavaland air.
+                    (ATMOS_OXYGEN, 4.0),   //
+                    (ATMOS_NITROGEN, 7.0), //
+                ])),
+                '#' => Wall,
+                _ => Space,
+            },
+            test_z,
+        );
+    }
+
+    // Air should exchange with Earth-like atmosphere.
+    #[test]
+    fn tick_affected_by_eartlike_air() {
+        let test_z = 12;
+
+        set_pattern(
+            &[
+                "###", //
+                "#X#", //
+                "###", //
+            ],
+            |c: char| match c {
+                'X' => Air(Box::new([
+                    (ATMOS_OXYGEN, 0.0),                       //
+                    (ATMOS_NITROGEN, 0.0),                     //
+                    (ATMOS_MODE, ATMOS_MODE_EARTHLIKE as f32), //
+                ])),
+                '#' => Wall,
+                _ => Space,
+            },
+            test_z,
+        );
+
+        tick();
+
+        expect_pattern(
+            &[
+                "###", //
+                "#Y#", //
+                "###", //
+            ],
+            |c: char| match c {
+                'Y' => Air(Box::new([
+                    // Gain 50% of standard air.
+                    (ATMOS_OXYGEN, STANDARD_OXYGEN_MOLES / 2.0), //
+                    (ATMOS_NITROGEN, STANDARD_NITROGEN_MOLES / 2.0), //
+                ])),
+                '#' => Wall,
+                _ => Space,
+            },
+            test_z,
+        );
+    }
+}
```

Comment:
Maybe split this file out?

Also might be worth porting this from RUSTG.
```rs
#[cfg(not(target_pointer_width = "32"))]
compile_error!("milla must be compiled for a 32-bit target");
```

Reply:
Will this make OD crap itself?

Reply:
> Will this make OD crap itself?

Yes but this is part of rustg and right now OD support is an insurance thing not a strive-towards goal.

Reply:
I'll take a better look at code organization later. For now, I'm happy leaving it as-is unless folks object.

The libraries we compile against already require 32-bit, so rust_g's compile_error is redundant.